File: Python/simple.py
import numpy as np
import time

if __name__ == "__main__":
  N = 100
  T = 100

  grid = [ [[0]*(N+2)]*(N+2) ]*2
  start = time.time()

  for t in xrange(1,T+1):
    write = t & 1
    read = 1 - write
    for i in xrange(1,N+1):
      for j in xrange(1,N+1):
        grid[write][i][j] = ( grid[read][i][j] +
                              grid[read][i+1][j] + grid[read][i][j+1] +
                              grid[read][i-1][j] + grid[read][i][j-1] ) / 5

  end = time.time()
  elapsed = end - start
  updates = ((N**2)*T)
  gflops = (updates*5)/1e9

  print "Cell Updates: {0}".format( updates )
  print "GFLOPS: {0}".format( gflops )
  print "Elapsed: {0}s".format( elapsed )
  print "Per Cell Update: {0}s".format( elapsed/updates )
  print "GFLOPS/s: {0}".format( gflops/elapsed )

File: Swift/src/simple.swift
import Foundation

var N = 4
var T = 3

var grid = Array<Array<Array<Double>>>( count: 2, repeatedValue: Array<Array<Double>>(count: N+2, repeatedValue: Array<Double>(count: N+2, repeatedValue: 0.0 ) ) )

let start = NSDate()

for t in 1...T{
  let write = t & 1
  let read = 1 - write

  for i in 1...N {
    for j in 1...N {
      // FIXME:
      // the ususal code
      // grid[write][i][j] = (grid[read][i][j] + grid[read][i+1][j] + grid[read][i][j+1] + grid[read][i-1][j] + grid[read][i][j-1] ) / 5
      // gives the error:
      // error: expression was too complex to be solved in reasonable time; consider breaking up the expression into distinct sub-expressions

      let a = grid[read][i][j] + grid[read][i+1][j]
      let b = grid[read][i][j+1] + grid[read][i-1][j]
      grid[write][i][j] = ( a + b + grid[read][i][j-1] ) / 5
    }
  }

}
let end = NSDate()
let elapsed = end.timeIntervalSinceDate( start )
print( "Elapsed: \(elapsed)s" )
